pkg/cqrs/cancellations.go

```diff
@@ -0,0 +1,42 @@
+package cqrs
+
+import (
+	"context"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/oklog/ulid/v2"
+)
+
+type CancellationReadWriter interface {
+	CancellationReader
+	CancellationWriter
+}
+
+// CancellationReader loads cancellations from a backing store.
+//
+// Note that this is intended for non-critical-path reads, eg. for dashboards
+// and management.  Critical path reader interfaces are defined within the
+// cancellation package.
+type CancellationReader interface {
+	// Cancellations returns all cancellations for a given workspace.
+	Cancellations(ctx context.Context, wsID uuid.UUID) ([]Cancellation, error)
+	CancellationsByFunction(ctx context.Context, wsID uuid.UUID, fnID uuid.UUID) ([]Cancellation, error)
+}
+
+type CancellationWriter interface {
+	// CreateCancellation writes a cancellation to the backing store.
+	CreateCancellation(ctx context.Context, c Cancellation) error
+}
+
+// Cancellation represents a cancellation of many function runs during the time specified.
+type Cancellation struct {
+	ID          ulid.ULID  `json:"id"`
+	WorkspaceID uuid.UUID  `json:"workspaceID"`
+	FunctionID  uuid.UUID  `json:"functionID"`
+	From        *time.Time `json:"from"`
+	To          time.Time  `json:"to"`
```

Comment:
**suggestion**: Should we use ISO8601 time strings here which could support millisecond precision? We often default to returning data in ISO8601 strings in other parts of the REST API instead of dealing with unix timestamp.s

Reply:
**question (dx)**: Is this the function start range? Perhaps it could be a little more obvious using params like:
```
started_after
started_before
```
These types of params could carry over to other endpoints that allow querying in the future like querying all runs by a time range of start time vs. end time (`ended_after`, `ended_before`). Obviously, with this endpoint, the ended would not be supported, but it would be consistent.

Reply:
`time.Time` in Golang is nanosecond precision by default, and when serialized to strings is sub-second precision: https://pkg.go.dev/time#Time.MarshalJSON.

I'll make the switch on `started_after` etc.

Reply:
Ok! I couldn't remember in my brain if this was serialized to an ISO string of a unix timestamp/int